Calvin Passmore

ECE 6470

## Homework 4

## Problem 1

You are synthesizing a chip composed of random logic with an average activity factor of 0.1.
You are using a standard cell process with an average switching capacitance of 450 pF/mm2.
Estimate the dynamic power consumption of your chip if it has an area of 70 mm2 and runs at 450 MHz at VDD=0.9V.

---

P<sub>dynamic</sub> = P<sub>switching</sub> + P<sub>shortcircuit</sub>

P<sub>shortcircuit</sub> &approx; 0

P<sub>switching</sub> = &alpha;C V<sub>DD</sub><sup>2</sup>f

In [3]:
# P switching
alpha = 0.1
C_A = 450e-12 # pf/mm2
A = 70 # mm2
C = C_A * A # pf
f = 450e6
Vdd = 0.9

alpha * C * Vdd**2 * f

1.1481750000000002

P<sub>dynamic</sub> = 1.148

<div style="break-after:page"></div>

## Problem 2

Determine the activity factor for the signal shown in Figure 5.34. The clock rate is 1 GHz

![](./Figure5_3.png)

The activity factor is 1/2 because out of 10 clock cycles, 5 of them are high, and 5 are low.

<div style="break-after:page"></div>

## Problem 3

Derive the switching probabilities in Table 5.1. 

| Gate | P<sub>Y</sub> |
|------|---------------|
| AND2 | P<sub>A</sub>P<sub>B</sub> |
| AND3 | P<sub>A</sub>P<sub>B</sub>P<sub>C</sub> |
| OR2  | 1 - <span style="text-decoration:overline">P</span><sub>A</sub><span style="text-decoration:overline">P</span><sub>B</sub> |
| NAND2 | 1 - P<sub>A</sub>P<sub>B</sub> |
| NOR2 |  <span style="text-decoration:overline">P</span><sub>A</sub> <span style="text-decoration:overline">P</span><sub>B</sub> |
| XOR2 | P<sub>A</sub><span style="text-decoration:overline">P</span><sub>B</sub> + <span style="text-decoration:overline">P</span><sub>A</sub>P<sub>B</sub>

![](./GatePropabilities.jpg)

<div style="break-after:page"></div>

## Problem 4

Construct a table similar to Table 5.2 for a 2-input NOR gate. (Assuming Vs is between A and B PMOS)

Ign = 6.3nA Igp = 0

Ioffn = 5.6nA Ioffp = 9.3nA

| Input State | I<sub>sub</sub> | I<sub>gate</sub> | I<sub>total</sub> | Vx |
|-------------|-----------------|------------------|-------------------|----|
| 00          |       11.2      |     0            |       11.2        | Stack Effect  |
| 01          |       9.3       |     6.3          |       15.6        | Vt |
| 10          |       9.3       |     6.3          |       15.6        | 1  |
| 11          | &approx;1       |     12.6         |       13.6        | 1  |